code-for-canada/project-thundercat
# [Email Response Multiselect] Addition of Email Response Multiselect

# Description

Added react-responsive-select to use drop down selects for To and CC fields
Added redux-mock-store, allowing us to mock redux state inside a test (used in EditActionDialog.test.js)

In a separate PR, I will:
- fix formatting of dropdowns
- try to remove some of the added css (if possible)

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Dependency change

## Screenshot

Add Email Response
![image](https://user-images.githubusercontent.com/2746350/56377128-a8d67280-61d7-11e9-92af-d9266dc3f65c.png)

Open To dropdown
![image](https://user-images.githubusercontent.com/2746350/56377141-b4299e00-61d7-11e9-916c-6898afd247f7.png)

Add Daniel
![image](https://user-images.githubusercontent.com/2746350/56377156-bf7cc980-61d7-11e9-8aa1-d9f912180f19.png)

Add Marina
![image](https://user-images.githubusercontent.com/2746350/56377217-e20ee280-61d7-11e9-8849-447082f7e6e3.png)

Open CC Dropdown and add Jack
![image](https://user-images.githubusercontent.com/2746350/56377250-01a60b00-61d8-11e9-9a64-810c3a6bf051.png)

Save response; open preview of response:
![image](https://user-images.githubusercontent.com/2746350/56377283-18e4f880-61d8-11e9-8a96-e6139f32f016.png)

Edit Response:
![image](https://user-images.githubusercontent.com/2746350/56377305-2306f700-61d8-11e9-87bd-2a2211157a2c.png)

Add Nancy:
![image](https://user-images.githubusercontent.com/2746350/56377328-30bc7c80-61d8-11e9-83a6-99f431295a0e.png)

Save, open preview:
![image](https://user-images.githubusercontent.com/2746350/56377362-40d45c00-61d8-11e9-80b8-d24039db4fd3.png)

# Testing

Applicable for all code changes.

Manual steps to reproduce this functionality:

1.  Go to the Inbox Tab of the test page
2,  Open a new email response
3.  Select options in the to/cc fields
4.  Save the response
5.  Open the preview
6.  Edit the response
7.  Ensure everything works as expected

# Checklist

Applicable for all code changes.

~~- [ ] I have made corresponding changes to the documentation~~
